File: leetcode-submissions/575_Distribute_Candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        bool flag=true;
        for(int  i=0;i<candyType.size();i++){
            for(int j=i+1;j<candyType.size();j++){
                if(candyType[i]!=candyType[j]){
                    flag=false;
                    break;
                }
            }
        }
        if(flag==true){
            return 1;
        }
        return (candyType.size()/2);
    }
};

File: leetcode-submissions/575_Distribute_Candies.ts
function distributeCandies(candyType: number[]): number {
    const mp:Map<number , number> = new Map();
    const n = candyType.length;
    for(const type of candyType){
        mp.set(type , (mp.get(type)! || 0 )+1);
    }
    const required = Math.floor(n/2) , types = mp.size;
    return Math.min(required   , types );
};